libs/rxidb/README.md

```diff
@@ -0,0 +1,220 @@
+
+function saveToDatabase<T>(key: string, value: T): void {
+  store$.pipe(setItem(key, value)).subscribe();
+}
+
+saveToDatabase('example_key', 'example_value');
+saveToDatabase('example_key2', 1337);
+```
+
+#### addItem()
+
+If you have an Object Store that is set to `autoIncrement`, you can use the `addItem` operator to add a new value to the store.
+
+```typescript
+import { connectIndexedDb, getObjectStore, addItem } from '@this-dot/rxidb';
+import { Observable } from 'rxjs';
+
+// Connect to an IndexedDB database (create if the database does not exist)
+const database$: Observable<IDBDatabase> = connectIndexedDb('example_database');
+// Create an Object Store or get an existing one with autoIncrement.
+const store$: Observable<IDBObjectStore> = database$.pipe(
+  getObjectStore('example_store', { autoIncrement: true })
+);
+
+function saveToDatabase<T>(key: string, value: T): void {
+  store$.pipe(addItem(key, value)).subscribe();
+}
+
+saveToDatabase('example_value');
+saveToDatabase(1337);
+```
+
+#### deleteItem()
+
+You can use the `deleteItem()` operator to delete a value from the store.
+
+```typescript
+import { connectIndexedDb, getObjectStore, setItem, deleteItem } from '@this-dot/rxidb';
+import { Observable } from 'rxjs';
+
+// Connect to an IndexedDB database (create if the database does not exist)
+const database$: Observable<IDBDatabase> = connectIndexedDb('example_database');
+// Create an Object Store or get an existing one.
+const store$: Observable<IDBObjectStore> = database$.pipe(getObjectStore('example_store'));
+
+store$.pipe(setItem('example_key', 'example_value')).subscribe();
+
+store$.pipe(deleteItem('example_key')).subscribe();
+```
+
+### Reading data from the database
+
+#### read()
+
+If you want to read a specific key's value from the database, you can use the `read()` operator.
+
+```typescript
+import { connectIndexedDb, getObjectStore, read, setItem, deleteItem } from '@this-dot/rxidb';
+import { Observable } from 'rxjs';
+
+// Connect to an IndexedDB database (create if the database does not exist)
+const database$: Observable<IDBDatabase> = connectIndexedDb('example_database');
+// Create an Object Store or get an existing one.
+const store$: Observable<IDBObjectStore> = database$.pipe(getObjectStore('example_store'));
+
+// set up the read stream.
+const exampleKey$: Observable<string> = store$.pipe(read('example_key'));
+
+exampleKey$.subscribe(console.log); // emits every time the value changes.
+
+// exampleKey$ will emit 4 times;
+store$
+  .pipe(
+    setItem('example_key', 'example_value_1'),
+    setItem('example_key', 'example_value_2'),
+    setItem('example_key', 'example_value_3'),
+    deleteItem('example_key')
+  )
+  .subscribe();
+```
+
+#### keys() & entries()
+
+You can get all the keys of a database using the `keys()` operator. This will return an `Observable` of all the keys in the store. With the `entries()` operator you can get all the values of the store.
```

Comment:
This confused me a few times - are "entries" and "values" the same thing in this case? Or does entries imply that it returns both the keys and values? If the latter is true, would be worth re-wording this to "With the `entries()` operator you can get all the key value pairs of the store"

Reply:
I renamed the `entries` method to `values` and I implemented the `entries` method to return key-value objects like

```
{
  key: 1,
  value: 'test'
}
```